tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php

```diff
@@ -156,4 +162,26 @@ public function createCustomerUsingCommand(string $customerReference, TableNode
         $id = $commandBus->handle($command);
         SharedStorage::getStorage()->set($customerReference, $id->getValue());
     }
+
+    /**
+     * @Given customer :reference does not exist
+     *
+     * @param string $reference
+     */
+    public function setNonExistingCustomerReference(string $reference): void
```

Comment:
I love what you did for these methods about non-existing entities Store the reference so that it can be used, it's much smarter than having a Behat statement dedicated to a specific action